### ISSUE TITLE
InternalOAuthError: failed to fetch user profile

### ISSUE THREAD

**Issue by manast** [auth0/passport-linkedin-oauth2#75]
So today, out of the blue this error appeared. On my app, that has been using linkedin singin since 2011. I have the latest version of passport-linkedin-oauth2, everything seems correct in the app settings in linkedin developer page... Any ideas what can produce this error? Seems somehow related to the API 1.0 deprecation, but what do we need in order to comply? since we were using OAuth2 already for years I did not think any action needs to be taken...
If somebody has a hint I really appreciate a prompt response, no customer can login in the system using linkedin which is quite a severe issue :/.

**Comment by manast** [comment]
ok. verified that the problem is with this module that has not updated to linkedin API 2.0 as reported in this issue:_ https://github.com/auth0/passport-linkedin-oauth2/issues/24
I have to say that I am *deeply disappointed* on the maintainers of this module. How is it possible that a change that has been communicated by linked in for many months has not yet arrived to this codebase?  This is breaking all production environments that uses this package, completely unacceptable. If you are unable to maintain it, please make it clear in the README.

**Comment by manast** [comment]
This fork has a fix for the problem: https://github.com/sokratisvidros/passport-linkedin-oauth2

**Comment by trungtangg** [comment]
Thanks for the fix. I've been racking my head all day trying to figure out why I couldn't fetch the profile.

**Comment by trungtangg** [comment]
> This fork has a fix for the problem: https://github.com/sokratisvidros/passport-linkedin-oauth2

IS there a reason why this only supports r_lite_profiles now?

**Comment by guix77** [comment]
I get same error with sokratisvidros's...

**Comment by ChungminPark** [comment]
@trungtangg Yeah I think you were right. All I have to do is change **r_basicprofile** to **r_liteprofile** to make it work.

**Comment by l5** [comment]
For me, the solution suggested in this comment solved the issue: https://github.com/auth0/passport-linkedin-oauth2/issues/102#issuecomment-1781370868

**Comment by mukeshsinghmanral** [comment]
**add "openid" in scope, it will work** 

passport.use(new LinkedinStrategy({
  clientID: process.env.LINKEDIN_CLIENT_ID,
  clientSecret: process.env.LINKEDIN_CLIENT_SECRET,
  callbackURL: "http://localhost:3000/auth/linkedin/callback",
  scope: ['email','profile',**'openid'**],
   
},
